api/src/main/resources/moodle.yaml

```diff
@@ -0,0 +1,163 @@
+#This file provide the end points for moodle interactions
+# MOodle API information can be found at :
+# https://docs.moodle.org/dev/Web_service_API_functions
+
+swagger: '2.0'
+
+info:
+  title: Moodle
+  description: |
+    Service for Moodle Interactions.
+  version: "0.1"
+
+host: "aoudev.nnlm.gov"
+schemes:
+  - "https"
+basePath: /webservice/rest
+
+produces:
+  - application/json
+
+##########################################################################################
+## PATHS
+##########################################################################################
+
+paths:
+  /server.php?wsfunction=core_user_get_users_by_field&moodlewsrestformat=json:
+    get:
+      summary: Retrieves user information from Moodle.
+      description: "Get user moodle information"
+      tags:
+        - Moodle
+      consumes:
+        - application/json
+      operationId: getMoodleId
+      parameters:
+        - name: field
+          in: query
+          description: Input field eg email on the basis of which information is received
+          type: string
+          required: false
+        - name: values[0]
+          in: query
+          description: Value of the input field
+          type: string
+          required: false
+        - name: wstoken
+          in: query
+          description: Token
+          type: string
+          required: true
+      responses:
+        200:
+          description: Response received
+          schema:
+            type: array
+            items:
+              $ref: '#/definitions/MoodleUserResponse'
+        400:
+          description: Bad request
+        500:
+          description: Moodle Internal Error
+      security:
+        - AdminSecurity: []
+
+  /server.php?wsfunction=core_badges_get_user_badges:
+   get:
+     summary: Get information of badges earned/completed by user.
+     description: "information of badges earned/completed by use"
+     tags:
+       - Moodle
+     consumes:
+       - application/json
+     operationId: getMoodleBadge
+     parameters:
+       - name: moodlewsrestformat
+         in: query
+         description: Format of the response which will be json
+         type: string
+         required: true
+       - name: userid
+         in: query
+         type: integer
+       - name: wstoken
+         in: query
+         description: Unique Token
+         type: string
+         required: true
+
+     responses:
+       200:
+         description: Badge /error response
+         schema:
+           $ref: '#/definitions/UserBadgeResponse'
+       400:
+         description: Bad request
+       500:
+         description: Moodle Internal Error
+     security:
+       - AdminSecurity: []
+
+##########################################################################################
+## DEFINITIONS
+##########################################################################################
+definitions:
+    MoodleUserResponse:
```

Comment:
These definitions are awfully blank! Obviously full API documentation isn't worth it (or even available anywhere on Moodle apparently), but it would be great to document at least the fields that we're reading from in our service file (MoodleUserResponse.id, and BadgeDetails.dateexpire)